tests/integration/integration_test.go

```diff
@@ -494,6 +494,7 @@ func TestConfigPaths(t *testing.T) {
 		"root[",
 		`root["nested]`,
 		"root.array[abc]",
+		"root.[0]",
 
```

Comment:
Can we check that ".foo" and ".[0]" also correctly get detected as invalid. Probably as part of the "First path segment must be a non-empty string." tests just below.